controllers/controllertest/mock_sagemaker_client.go

```diff
@@ -274,9 +274,9 @@ func (m *MockSageMakerClientBuilder) AddDescribeEndpointErrorResponse(code strin
 }
 
 // Add a DescribeTrainingJob error response to the client.
-func (m *MockSageMakerClientBuilder) AddDescribeTransformJobErrorResponse(code string, statusCode int, reqId string) *MockSageMakerClientBuilder {
+func (m *MockSageMakerClientBuilder) AddDescribeTransformJobErrorResponse(code string, statusCode int, reqId, message string) *MockSageMakerClientBuilder {
 	m.responses.PushBack(describeTransformJobResponse{
-		err:  awserr.NewRequestFailure(awserr.New(code, "mock error message", fmt.Errorf(code)), statusCode, reqId),
+		err:  awserr.NewRequestFailure(awserr.New(code, message, fmt.Errorf(code)), statusCode, reqId),
```

Comment:
Does this need to have a default value to not break existing tests?

Reply:
Nope, this is there because when I first wrote this I did not understand difference between code and message.

Reply:
You'll have to explain to me the difference as well, then haha

Reply:
Code is like a java classname, e.g. "ThrottleException", or "ValidationException". Message is the particular instance type, e.g. "No cross-account roles". SageMaker API does not use traditional HTTP codes (a lot of errors are multiplexed into HTTP 400), so we have to use both of these to extract the error cause.